package zippylog.device.streamer;

message Create {
	optional bytes id = 1;
}

message Destroy {
	optional bytes id = 1;
}

message SubscriptionExpired {
	optional bytes id = 1;
	optional bytes subscription = 2;
}

message ReceiveKeepalive {
	optional bytes id = 1;
	optional bytes subscription = 2;
}

message SubscriptionRenewedFromKeepalive {
	optional bytes id = 1;
	optional bytes subscription = 2;
}

message RejectKeepaliveUnknownSubscription {
	optional bytes id = 1;
	optional bytes subscription = 2;
}

message ErrorRenewingSubscription {
	optional bytes id = 1;
	optional bytes subscription = 2;
}
